pkl-core/src/main/java/org/pkl/core/ast/type/ResolveDeclaredTypeNode.java

```
@@ -73,11 +74,36 @@ protected VmTyped getImport(
     var result = module.getCachedValue(importName);
     if (result == null) {
       result = callNode.call(member.getCallTarget(), module, module, importName);
+
+      var importedModule = (VmTyped) result;
+      if (importedModule.isNotInitialized()
+          && importedModule.isModuleObject()
+          && importedModule.getModuleInfo().isAmend()) {
+        // this is an amending module. Try to find the prototype
+        var prototypeModule = findPrototypeModule(importedModule);
+        if (prototypeModule != null) {
+          result = prototypeModule;
+        }
+      }
+
       module.setCachedValue(importName, result);
     }
     return (VmTyped) result;
   }
 
+  private @Nullable VmTyped findPrototypeModule(VmTyped amendingModule) {
+    try {
+      var moduleInfo = amendingModule.getModuleInfo();
+      var amendedModuleKey = moduleInfo.getAmendedModuleKey();
+
+      if (amendedModuleKey != null) {
+        return VmLanguage.get(null).loadModule(amendedModuleKey);
+      }
+    } catch (Exception ignored) {
+    }
```

Why ignore all exceptions here?

---

Yeah, if there's an error here it should just bubble up. Removed the try.